C/plugins/storage/sqlite/common/readings.cpp

```diff
@@ -186,28 +200,60 @@ bool Connection::aggregateQuery(const Value& payload, string& resultSet)
 	// Add sum
 	sql.append("\"sum\" : ' || sum(theval) || '}' AS resd ");
 
+	if (size < 1)
+	{
+		// Add max(user_ts)
+		sql.append(", max(" + timeColumn + ") AS " + timeColumn + " ");
+	}
+
 	// subquery
 	sql.append("FROM ( SELECT asset_code, ");
 	sql.append(timeColumn);
-	sql.append(", datetime(");
+
+	if (size >= 1)
+	{
+		sql.append(", datetime(");
+	}
+	else
+	{
+		sql.append(", (");
+	}
+
+	// Size formatted string
+	string size_format;
+	if (fmod(size, 1.0) == 0.0)
+	{
+		size_format = to_string(int(size));
+	}
+	else
+	{
+		size_format = to_string(size);
+	}
 
 	// Add timebucket size
-	if (size > 1)
+	if (size != 1)
 	{
-		sql.append(to_string(size));
-		sql.append(" * round(strftime('%s', ");
+		sql.append(size_format);
+		sql.append(" * round((julianday(");
 		sql.append(timeColumn);
-		sql.append(") / ");
-		sql.append(to_string(size));
-		sql.append(", 6), 'unixepoch') ");
+		sql.append(") - 2440587.5) * 86400.0 / ");
```

Comment:
It might be good to use #define's or constants here, just so everybody is clear what 2440587.5 and 86400.0 re - my guess is they are some important numbers!